### PR TITLE
Support wide operations in queues

### PR DESCRIPTION
**Pull request by papaef**
Dear Wilson & Contributors to Verilator,

Since this is the first time I communicate with you I would like to thank you for this amazing effort all these years. Verilator has become my beloved simulator :)

I came across an issue while testing the Systemverilog queues and found that wide arguments (>64 bits) where not supported in the current implementation and the code could not compile.
I fixed the issue and also provide a test-case in this pull request.

Let me know if the fix looks OK to you.

Thanks a lot and best regards,
Vassilis Papaefstathiou

### CODE REVIEW

File: src/V3EmitC.cpp [hunk 1]
```diff
@@ -1648,7 +1662,11 @@ class EmitCImp : EmitCStmts {
         } else if (AstDynArrayDType* adtypep = VN_CAST(dtypep, DynArrayDType)) {
             return emitVarResetRecurse(varp, adtypep->subDTypep(), depth + 1, ".atDefault()");
```
[the remark this answers is not in the document]
Comment (papaef) [reply src/V3EmitC.cpp]: Sure, I just updated the code to fix the dynamic arrays too. I also updated the associated tests (t_queue.v and t_dynarray.v)

File: src/V3EmitC.cpp [hunk 2]
```diff
@@ -351,10 +351,24 @@ class EmitCStmts : public EmitCBaseVisitor {
...
             if (comma) puts(", ");
-            iterate(subnodep);
+            // handle wide arguments to the Queues
+            if (VN_IS(nodep->fromp()->dtypep(), QueueDType) && subnodep->dtypep()->isWide() ) {
+                putbs("VL_CVT_W_A(");
+                iterate(subnodep);
+                puts(", ");
+                iterate(nodep->fromp());
+                putbs(".atDefault()");  // Not accessed; only to get the proper type of values
+                puts(")");
```
[the remark this answers is not in the document]
Comment (papaef) [reply src/V3EmitC.cpp]: Yes, for the queues I looked the code for associative arrays and modified it appropriately. 
I followed your suggestion and created the emitCvtWideArray function and replaced the inline code.

Let me know if it looks OK now.

**Comment by wsnyder** [comment]
Thanks for your first contribution! Hope to see more!

One minor note for next time is to appease the Travis tests you should fix your github username so it matches your .git/config username (see failing test).  I pushed it manually to resolve this.

**Comment by papaef** [comment]
Thanks Wilson! It is my pleasure to contribute to Verilator. I will definitely contribute more when possible. 

I tried to fix the issue with Travis tests in the previous commit but it seems that Github uses the information from the website profile. My local git config is correct. I have updated my details on the web profile and hope that next time the script will not complain.

Cheers!
Vassilis

**Pull request closed without merging**